app/routes/orders/conversation.js

```diff
@@ -1,3 +1,21 @@
 import detail from './detail';
+import Ember from 'ember';
 
-export default detail.extend();
+export default detail.extend({
+  model(params){
+    var designation = this.store.peekRecord("designation", params.order_id, {
+      reload: true }) || this.store.findRecord('designation', params.order_id);
+    return Ember.RSVP.hash({
+      designation: designation,
```

Comment:
This won't work in case if it fetches designation from API.  As it's asynchronous . 
You'll need nested promise

Reply:
I think replacing it with `RSVP.all` should do the trick. It'll resolve the promises, and those that aren't promises will just pass through

Reply:
I think adding this fetching `designation` directly inside the `RSVP.hash` should work , it will resolve all request altogether ?